app/export.go

```diff
@@ -43,6 +44,18 @@ func (app *GaiaApp) ExportAppStateAndValidators(
 	}
 
 	validators, err := staking.WriteValidators(ctx, app.StakingKeeper)
+	if err != nil {
+		return servertypes.ExportedApp{}, err
+	}
+	sort.SliceStable(validators, func(i, j int) bool {
+		return validators[i].Power > validators[j].Power
+	})
+	// we have to trim this to only active consensus validators
+	maxVals := app.ProviderKeeper.GetMaxProviderConsensusValidators(ctx)
```

Comment:
This would delete validators that are allowed to validate consumer chains but are not in the active validator set of the hub.

Would you mind expanding on the issue?

Reply:
Sure. [This](https://github.com/cosmos/cosmos-sdk/blob/ec58adf8e2fa4149c093d871d2fd398ca508525a/baseapp/abci.go#L111-L116) assertion fails when you're importing a chain right now. This is because the provider keeper only answered with the consensus validators.

Would the validators actually be deleted? I was expecting them to still be in the app_state for `staking` so that once the chain starts up they get added back to the set, but I may be wrong

Reply:
This command if for exporting when creating a new `genesis.json` if I'm not mistaken.

In that case, yes, they would be deleted and not available in the new chain.